src/app/util/ember-compatibility-functions.cpp

```diff
@@ -210,222 +221,254 @@ bool ServerClusterCommandExists(const ConcreteCommandPath & aCommandPath)
-    EmberAfStatus emberStatus = emAfReadOrWriteAttribute(&record, &metadata, attributeData, sizeof(attributeData),
-                                                         /* write = */ false);
-
-    if (emberStatus == EMBER_ZCL_STATUS_SUCCESS)
-    {
-        EmberAfAttributeType attributeType = metadata->attributeType;
-        bool isNullable                    = metadata->IsNullable();
-        writer                             = attributeDataIBBuilder.GetWriter();
-        VerifyOrReturnError(writer != nullptr, CHIP_NO_ERROR);
-        TLV::Tag tag = TLV::ContextTag(to_underlying(AttributeDataIB::Tag::kData));
-        switch (BaseType(attributeType))
-        {
-        case ZCL_NO_DATA_ATTRIBUTE_TYPE: // No data
-            ReturnErrorOnFailure(writer->PutNull(tag));
-            break;
-        case ZCL_BOOLEAN_ATTRIBUTE_TYPE: // Boolean
-            ReturnErrorOnFailure(attributeBufferToNumericTlvData<bool>(*writer, isNullable));
-            break;
-        case ZCL_INT8U_ATTRIBUTE_TYPE: // Unsigned 8-bit integer
-            ReturnErrorOnFailure(attributeBufferToNumericTlvData<uint8_t>(*writer, isNullable));
-            break;
-        case ZCL_INT16U_ATTRIBUTE_TYPE: // Unsigned 16-bit integer
+        AttributeAccessInterface * attrOverride = findAttributeAccessOverride(aPath.mEndpointId, aPath.mClusterId);
+        if (attrOverride != nullptr)
         {
-            ReturnErrorOnFailure(attributeBufferToNumericTlvData<uint16_t>(*writer, isNullable));
-            break;
-        }
-        case ZCL_INT32U_ATTRIBUTE_TYPE: // Unsigned 32-bit integer
-        {
-            ReturnErrorOnFailure(attributeBufferToNumericTlvData<uint32_t>(*writer, isNullable));
-            break;
-        }
-        case ZCL_INT64U_ATTRIBUTE_TYPE: // Unsigned 64-bit integer
-        {
-            ReturnErrorOnFailure(attributeBufferToNumericTlvData<uint64_t>(*writer, isNullable));
-            break;
-        }
-        case ZCL_INT8S_ATTRIBUTE_TYPE: // Signed 8-bit integer
-        {
-            ReturnErrorOnFailure(attributeBufferToNumericTlvData<int8_t>(*writer, isNullable));
-            break;
-        }
-        case ZCL_INT16S_ATTRIBUTE_TYPE: // Signed 16-bit integer
-        {
-            ReturnErrorOnFailure(attributeBufferToNumericTlvData<int16_t>(*writer, isNullable));
-            break;
+            // TODO: We should probably clone the writer and convert failures here
+            // into status responses, unless our caller already does that.
+
+            AttributeValueEncoder valueEncoder(writer, aSubjectDescriptor.fabricIndex);
+            ReturnErrorOnFailure(attrOverride->Read(aPath, valueEncoder));
+
+            auto accessControlStatus = CheckAccessControl(aSubjectDescriptor, aPath.mEndpointId, aPath.mClusterId, /* write = */ false);
+            if (accessControlStatus != Protocols::InteractionModel::Status::Success)
+            {
+                status = accessControlStatus;
```

Comment:
This doesn't seem like it correctly handles wildcard reads where upon path expansion, instead of return back errors, paths that have errors in access control checks are discarded. Follow-up fine assuming you're still in transitional mode, but please track this.

Reply:
There's an issue for it. This PR doesn't do everything.